### ISSUE TITLE
[Bug] Undefined array Key and deprecated 

### ISSUE THREAD

**Issue by gOOvER** [hestiacp/hestiacp#2822]
### Describe the bug

i have a lot of undefind Errors in my log and also Deprcated Warnings

- PHP Warning:  Undefined array key "v_debug_mode" in /usr/local/hestia/web/edit/server/index.php on line 422
- PHP Warning:  Undefined array key "v_experimental_features" in /usr/local/hestia/web/edit/server/index.php on line 451
- PHP Warning:  Undefined array key "v_policy_user_view_suspended" in /usr/local/hestia/web/edit/server/index.php on line 462
- PHP Warning:  Undefined array key "v_policy_user_view_suspended" in /usr/local/hestia/web/edit/server/index.php on line 464
- PHP Warning:  Undefined array key "v_pgsql_url" in /usr/local/hestia/web/edit/server/index.php on line 620
- PHP Warning:  Undefined array key "v_policy_user_view_suspended" in /usr/local/hestia/web/edit/server/index.php on line 1168" while reading response header from upstream, client: x.x.x.x, server: _, request: "POST /edit/server/ HTTP/1.1", upstream: "fastcgi://unix:/var/run/hestia-php.sock:", host: "xx.xx.xx:8083", referrer: "https://xxx.xxx.xxx:8083/edit/server/"

- PHP Deprecated:  escapeshellarg(): Passing null to parameter #1 ($arg) of type string is deprecated in /usr/local/hestia/web/edit/server/index.php on line 464

 

### Tell us how to replicate the bug

See nginx error.log with php 8.1 activated

### Which components are affected by this bug?

Control Panel Backend

### Hestia Control Panel Version

1.6.6

### Operating system

Debian 11

### Log capture

```shell
2022/08/05 14:21:05 [error] 715221#0: *52 FastCGI sent in stderr: "PHP message: PHP Warning:  Undefined array key "v_nginx_cache_duration" in /usr/local/hestia/web/edit/web/index.php on line 255" while reading response header from upstream, client: 84.115.227.203, server: _, request: "POST /edit/web/?domain=domain.tld&token=eba80607a03fb7988e59f89ea6298848 HTTP/1.1", upstream: "fastcgi://unix:/var/run/hestia-php.sock:", host: "my.domain.tld:8083", referrer: "https://my.domain.tld:8083/edit/web/?domain=domain.tld&token=eba80607a03fb7988e59f89ea6298848"
2022/08/05 14:21:05 [error] 715221#0: *52 FastCGI sent in stderr: "PHP message: PHP Warning:  Undefined variable $v_redirect_custom in /usr/local/hestia/web/templates/pages/edit_web.html on line 178" while reading response header from upstream, client: 84.115.227.203, server: _, request: "GET /edit/web/?domain=domain.tld HTTP/1.1", upstream: "fastcgi://unix:/var/run/hestia-php.sock:", host: "my.domain.tld:8083", referrer: "https://my.domain.tld:8083/edit/web/?domain=domain.tld&token=eba80607a03fb7988e59f89ea6298848"
2022/08/05 14:22:30 [error] 715221#0: *86 FastCGI sent in stderr: "PHP message: PHP Warning:  Undefined variable $v_redirect_custom in /usr/local/hestia/web/templates/pages/edit_web.html on line 178" while reading response header from upstream, client: 84.115.227.203, server: _, request: "GET /edit/web/?domain=domain.tld&token=eba80607a03fb7988e59f89ea6298848 HTTP/1.1", upstream: "fastcgi://unix:/var/run/hestia-php.sock:", host: "my.domain.tld:8083", referrer: "https://my.domain.tld:8083/list/web/"
2022/08/05 14:23:12 [error] 715221#0: *111 FastCGI sent in stderr: "PHP message: PHP Warning:  Undefined array key "v_debug_mode" in /usr/local/hestia/web/edit/server/index.php on line 422PHP message: PHP Warning:  Undefined array key "v_experimental_features" in /usr/local/hestia/web/edit/server/index.php on line 451PHP message: PHP Warning:  Undefined array key "v_policy_user_view_suspended" in /usr/local/hestia/web/edit/server/index.php on line 462PHP message: PHP Warning:  Undefined array key "v_policy_user_view_suspended" in /usr/local/hestia/web/edit/server/index.php on line 464PHP message: PHP Deprecated:  escapeshellarg(): Passing null to parameter #1 ($arg) of type string is deprecated in /usr/local/hestia/web/edit/server/index.php on line 464PHP message: PHP Warning:  Undefined array key "v_pgsql_url" in /usr/local/hestia/web/edit/server/index.php on line 620PHP message: PHP Warning:  Undefined array key "v_policy_user_view_suspended" in /usr/local/hestia/web/edit/server/index.php on line 1168" while reading response header from upstream, client: 84.115.227.203, server: _, request: "POST /edit/server/ HTTP/1.1", upstream: "fastcgi://unix:/var/run/hestia-php.sock:", host: "my.domain.tld:8083", referrer: "https://my.domain.tld:8083/edit/server/"
2022/08/05 14:23:25 [error] 715221#0: *111 FastCGI sent in stderr: "PHP message: PHP Warning:  Undefined variable $v_redirect_custom in /usr/local/hestia/web/templates/pages/edit_web.html on line 178" while reading response header from upstream, client: 84.115.227.203, server: _, request: "GET /edit/web/?domain=domain.tld&token=eba80607a03fb7988e59f89ea6298848 HTTP/1.1", upstream: "fastcgi://unix:/var/run/hestia-php.sock:", host: "my.domain.tld:8083", referrer: "https://my.domain.tld:8083/list/web/"
2022/08/05 14:23:41 [error] 715221#0: *111 FastCGI sent in stderr: "PHP message: PHP Warning:  Undefined array key "v_nginx_cache_duration" in /usr/local/hestia/web/edit/web/index.php on line 255" while reading response header from upstream, client: 84.115.227.203, server: _, request: "POST /edit/web/?domain=domain.tld&token=eba80607a03fb7988e59f89ea6298848 HTTP/1.1", upstream: "fastcgi://unix:/var/run/hestia-php.sock:", host: "my.domain.tld:8083", referrer: "https://my.domain.tld:8083/edit/web/?domain=domain.tld&token=eba80607a03fb7988e59f89ea6298848"
2022/08/05 14:23:41 [error] 715221#0: *111 FastCGI sent in stderr: "PHP message: PHP Warning:  Undefined variable $v_redirect_custom in /usr/local/hestia/web/templates/pages/edit_web.html on line 178" while reading response header from upstream, client: 84.115.227.203, server: _, request: "GET /edit/web/?domain=domain.tld HTTP/1.1", upstream: "fastcgi://unix:/var/run/hestia-php.sock:", host: "my.domain.tld:8083", referrer: "https://my.domain.tld:8083/edit/web/?domain=domain.tld&token=eba80607a03fb7988e59f89ea6298848"
2022/08/05 14:32:48 [error] 715221#0: *170 FastCGI sent in stderr: "PHP message: PHP Warning:  Undefined variable $v_redirect_custom in /usr/local/hestia/web/templates/pages/edit_web.html on line 178" while reading response header from upstream, client: 84.115.227.203, server: _, request: "GET /edit/web/?domain=domain.tld&token=eba80607a03fb7988e59f89ea6298848 HTTP/1.1", upstream: "fastcgi://unix:/var/run/hestia-php.sock:", host: "my.domain.tld:8083", referrer: "https://my.domain.tld:8083/list/web/"
2022/08/05 14:36:55 [error] 715221#0: *249 FastCGI sent in stderr: "PHP message: PHP Warning:  Undefined variable $v_redirect_custom in /usr/local/hestia/web/templates/pages/edit_web.html on line 178" while reading response header from upstream, client: 84.115.227.203, server: _, request: "GET /edit/web/?domain=domain.tld&token=eba80607a03fb7988e59f89ea6298848 HTTP/1.1", upstream: "fastcgi://unix:/var/run/hestia-php.sock:", host: "my.domain.tld:8083", referrer: "https://my.domain.tld:8083/list/web/"
2022/08/05 14:49:31 [error] 715221#0: *285 FastCGI sent in stderr: "PHP message: PHP Warning:  Undefined variable $v_proxy in /usr/local/hestia/web/templates/pages/add_web.html on line 189PHP message: PHP Warning:  Undefined variable $v_proxy in /usr/local/hestia/web/templates/pages/add_web.html on line 218" while reading response header from upstream, client: 84.115.227.203, server: _, request: "GET /add/web/ HTTP/1.1", upstream: "fastcgi://unix:/var/run/hestia-php.sock:", host: "my.domain.tld:8083", referrer: "https://my.domain.tld:8083/list/web/"
2022/08/05 15:00:21 [error] 715221#0: *305 FastCGI sent in stderr: "PHP message: PHP Warning:  Undefined variable $v_redirect_custom in /usr/local/hestia/web/templates/pages/edit_web.html on line 178" while reading response header from upstream, client: 84.115.227.203, server: _, request: "GET /edit/web/?domain=domain.tld&token=eba80607a03fb7988e59f89ea6298848 HTTP/1.1", upstream: "fastcgi://unix:/var/run/hestia-php.sock:", host: "my.domain.tld:8083", referrer: "https://my.domain.tld:8083/list/web/"
2022/08/05 15:00:33 [error] 715221#0: *305 FastCGI sent in stderr: "PHP message: PHP Warning:  Undefined array key "v_nginx_cache_duration" in /usr/local/hestia/web/edit/web/index.php on line 255" while reading response header from upstream, client: 84.115.227.203, server: _, request: "POST /edit/web/?domain=domain.tld&token=eba80607a03fb7988e59f89ea6298848 HTTP/1.1", upstream: "fastcgi://unix:/var/run/hestia-php.sock:", host: "my.domain.tld:8083", referrer: "https://my.domain.tld:8083/edit/web/?domain=domain.tld&token=eba80607a03fb7988e59f89ea6298848"
2022/08/05 15:00:33 [error] 715221#0: *305 FastCGI sent in stderr: "PHP message: PHP Warning:  Undefined variable $v_redirect_custom in /usr/local/hestia/web/templates/pages/edit_web.html on line 178" while reading response header from upstream, client: 84.115.227.203, server: _, request: "GET /edit/web/?domain=domain.tld HTTP/1.1", upstream: "fastcgi://unix:/var/run/hestia-php.sock:", host: "my.domain.tld:8083", referrer: "https://my.domain.tld:8083/edit/web/?domain=domain.tld&token=eba80607a03fb7988e59f89ea6298848"
2022/08/05 15:55:37 [error] 715221#0: *400 FastCGI sent in stderr: "PHP message: PHP Warning:  Undefined variable $v_redirect_custom in /usr/local/hestia/web/templates/pages/edit_web.html on line 178" while reading response header from upstream, client: 84.115.227.203, server: _, request: "GET /edit/web/?domain=domain.tld&token=eba80607a03fb7988e59f89ea6298848 HTTP/1.1", upstream: "fastcgi://unix:/var/run/hestia-php.sock:", host: "my.domain.tld:8083", referrer: "https://my.domain.tld:8083/list/web/"
2022/08/06 07:37:29 [error] 715221#0: *488 FastCGI sent in stderr: "PHP message: PHP Warning:  Undefined variable $v_redirect_custom in /usr/local/hestia/web/templates/pages/edit_web.html on line 178" while reading response header from upstream, client: 84.115.227.203, server: _, request: "GET /edit/web/?domain=domain.tld&token=f5d9ea22781bb3f39e6d36e0ca5253c2 HTTP/1.1", upstream: "fastcgi://unix:/var/run/hestia-php.sock:", host: "my.domain.tld:8083", referrer: "https://my.domain.tld:8083/list/web/"
2022/08/06 07:37:35 [error] 715221#0: *489 FastCGI sent in stderr: "PHP message: PHP Warning:  Undefined variable $v_redirect_custom in /usr/local/hestia/web/templates/pages/edit_web.html on line 178" while reading response header from upstream, client: 84.115.227.203, server: _, request: "GET /edit/web/?domain=domain.tld&token=f5d9ea22781bb3f39e6d36e0ca5253c2 HTTP/1.1", upstream: "fastcgi://unix:/var/run/hestia-php.sock:", host: "my.domain.tld:8083", referrer: "https://my.domain.tld:8083/list/web/"
2022/08/06 08:00:31 [error] 715221#0: *572 FastCGI sent in stderr: "PHP message: PHP Warning:  Undefined array key "v_debug_mode" in /usr/local/hestia/web/edit/server/index.php on line 422PHP message: PHP Warning:  Undefined array key "v_experimental_features" in /usr/local/hestia/web/edit/server/index.php on line 451PHP message: PHP Warning:  Undefined array key "v_policy_user_view_suspended" in /usr/local/hestia/web/edit/server/index.php on line 462PHP message: PHP Warning:  Undefined array key "v_policy_user_view_suspended" in /usr/local/hestia/web/edit/server/index.php on line 464PHP message: PHP Deprecated:  escapeshellarg(): Passing null to parameter #1 ($arg) of type string is deprecated in /usr/local/hestia/web/edit/server/index.php on line 464PHP message: PHP Warning:  Undefined array key "v_pgsql_url" in /usr/local/hestia/web/edit/server/index.php on line 620PHP message: PHP Warning:  Undefined array key "v_policy_user_view_suspended" in /usr/local/hestia/web/edit/server/index.php on line 1168" while reading response header from upstream, client: 84.115.227.203, server: _, request: "POST /edit/server/ HTTP/1.1", upstream: "fastcgi://unix:/var/run/hestia-php.sock:", host: "my.domain.tld:8083", referrer: "https://my.domain.tld:8083/edit/server/"
```

**Comment by jaapmarcus** [comment]
Should be save to ignore... 

PHP has be a lot more strict when we switched to php8.1 for hestia

**Comment by turbopixel** [comment]
Hi @jaapmarcus 
I'll go through the files and fix the PHP warnings and make the code PHP 8.1 compliant, if you want it.

**Comment by jaapmarcus** [comment]
> Hi @jaapmarcus 
> 
> I'll go through the files and fix the PHP warnings and make the code PHP 8.1 compliant, if you want it. 

Please go a head

**Comment by ScIT-Raphael** [comment]
As @jaapmarcus already wrote, for now save to ignore. We're working already on rewrites on the interface which should make it step by step more php 8.1 compliant. I'll go to close the issue.
